packages/slice-machine/src/tracker.ts

```diff
@@ -0,0 +1,176 @@
+import type { Analytics as ClientAnalytics } from "@segment/analytics-next";
+import { AnalyticsBrowser } from "@segment/analytics-next";
+import { Frameworks } from "@slicemachine/core/build/src/models";
+import { LibraryUI } from "@models/common/LibraryUI";
+
+// These events should be sync with the tracking Plan on segment.
+type AllSliceMachineEventType = EventType | ContinueOnboardingType;
+
+enum EventType {
+  Review = "SliceMachine Review",
+  OnboardingStart = "SliceMachine Onboarding Start",
+  OnboardingSkip = "SliceMachine Onboarding Skip",
+  SlicePreviewSetup = "SliceMachine Slice Preview Setup",
+  SlicePreviewOpen = "SliceMachine Slice Preview Open",
+}
+
+export enum ContinueOnboardingType {
+  OnboardingContinueIntro = "SliceMachine Onboarding Continue Screen Intro",
+  OnboardingContinueScreen1 = "SliceMachine Onboarding Continue Screen 1",
+  OnboardingContinueScreen2 = "SliceMachine Onboarding Continue Screen 2",
+  OnboardingContinueScreen3 = "SliceMachine Onboarding Continue Screen 3",
+}
+
+let _client: ClientAnalytics | null = null;
+let _isTrackingActive = true;
+let _repoName: string | null = null;
+
+/*
+ * TEMPORARY
+ * Today, we need to have access to the cookie to get the anonymousID from segment but
+ * we want to log that event only at start time. Because of the necessity to access the cookies,
+ * we can't put that code on the start script just yet.
+ * We need to define a potential alternative for this event.
+ * In the meantime, this flag prevents the event flood
+ */
+let _temp_first_start_flag = true;
+
+/** Private methods **/
+
+// Native client event method (don't expose these functions)
+const _trackEvent = (
+  eventType: AllSliceMachineEventType,
+  attributes: Record<string, unknown> = {}
+): void => {
+  if (!_isTrackingPossible(_client)) {
```

Comment:
you cannot make this check everywhere, either the tracker is possibly null or you wrap the whole thing that will process each call to the tracker and drop them if the client is null or execute it but you can't make that check for every single track